main/glossary/index.md

```diff
@@ -602,6 +602,27 @@ Secure ECMAScript has been renamed to [Hardened JavaScript](#hardened-javascript
 
 An imaginary currency Agoric documentation uses in examples.
 
+## Smart Contract
+
+A smart contract is a contract-​like arrangement expressed in code, where the behavior of the 
+program enforces the terms of the contract. Smart contracts are able to solve some of the
+fundamental social problems that have traditionally been solved by legal contracts. For example,
+commerce often involves an exchange of one thing for another. However, the party that transfers
+their assets first leaves themselves vulnerable to the other party’s opportunism: the second party
+can simply walk away from the exchange with both parties’ assets. 
+
+Smart contracts solve this problem of opportunism during an exchange by providing an 
+[escrow mechanism](#escrow) for the assets from both parties, only dispersing the assets after both
+parties have contributed. There is no need for a human intermediary to perform the escrow; the smart
+contract itself holds the assets. Because the behavior of the smart contract is fully described by
+its code, and that code runs on a decentralized blockchain with high integrity execution, the
+parties can be assured that the code describes exactly what will happen. As such, smart contracts
+hold the promise of being a market-​based solution for the opportunism problem that does not rely on
+State intervention.
```

Comment:
I'd rather skip the commentary on social problems and State intervention in this context:

```suggestion
A smart contract is a contract-​like arrangement expressed in code, where the behavior of the 
program enforces the terms of the contract.
```

In general, the glossary shouldn't have novel content. The terms should be defined in sections that discuss them. Each glossary entry should be a short gloss with a link to the relevant section.

Here we have a link to `#offer-safety`, which is perhaps good enough. But better would be a link to [Smart Contract Basics](https://docs.agoric.com/guides/zoe/contract-basics.html#smart-contract-basics).